src/paint.js

```diff
@@ -297,12 +297,12 @@ function setupPaint({ $, qlik }) {
           .domain(scaleDomain)
           .range(colors);
 
-        gridSize = Math.floor((width - margin.left - margin.right) / gridDivider);
+        gridSize = Math.floor((width - margin.left - margin.right) / dim2keys.length);
         if (gridSize <= thresholds.minimum){
           gridSize = thresholds.minimum;
         }
         const thresholdClasses = getThresholdClasses(gridSize);
-        legendElementWidth = Math.floor((gridSize * gridDivider) / (colorScale.quantiles().length + 1));
+        legendElementWidth = Math.floor((gridSize * dim2keys.length) / (colorScale.quantiles().length + 1));
```

Comment:
Why have you chosen to ignore maxGridColums here altogether ?

Reply:
gridDivider is calculated like below 
var gridDivider = Math.max(maxGridColums * 1, dim2keys.length);
 
According to my understanding leaving margin-left and margin-right whatever space left should be divided by the number of columns in the data
But whenever we take grid divider as Math.max(maxGridColums * 1, dim2keys.length); it always comes as 18( maxGridColums value declared in define.js)  if dim2Keys is less than 18 
so grid size was not calculated according to the date so blank space was coming  in the left

If we calculate grid size based on the number of columns in the data (which is dim2keys) data will fit the page properly
correct me if I am wrong

Reply:
Thanks for the explanation @chaitraraoqlik , sounds reasonable.